### PR TITLE
[P] [RELNOTES] Add ability to visualize wrapped models with plot_model function

### PR DESCRIPTION
**Pull request by yoks**
### Summary
Added two new optional arguments to `plot_model` function in `vis_utils.py`, to be able to visualize nested (wrapped) models. Also adds `dpi` param to have ability to produce high resolution graphs.

For example giving this model:
```python
sentence_input = Input(shape=(2, 3), dtype='float32', name="input2")
l_lstm = Bidirectional(LSTM(16))(sentence_input)
sent_encoder = Model(sentence_input, l_lstm)
review_input = Input(shape=(5, 2, 3), dtype='float32')
review_encoder = TimeDistributed(sent_encoder)(review_input)
l_lstm_sent = LSTM(16)(review_encoder)
preds = Dense(5, activation='softmax')(l_lstm_sent)
model = Model(review_input, preds)
vis_utils.plot_model(model, to_file='model3.png', show_shapes=True,
                     expand_nested=True, dpi=300)
```
Will produce:
model3

While calling it without new arguments will produce plot as before:
```python
vis_utils.plot_model(model, to_file='model3.png', show_shapes=True)
```
model3

### Related Issues
#5937

### PR Overview

- [x] This PR requires new unit tests [y/n] (make sure tests are included)
- [x] This PR requires to update the documentation [y/n] (make sure the docs are up-to-date)
- [x] This PR is backwards compatible [y/n]
- [x] This PR changes the current API [y/n] (all API changes need to be approved by fchollet)

### CODE REVIEW

**Comment by gabrieldemarmiesse** [comment]
There is a API change. Pinging @fchollet .

File: docs/templates/visualization.md [hunk 1]
```diff
@@ -10,10 +10,12 @@ from keras.utils import plot_model
 plot_model(model, to_file='model.png')
 ```
 
-`plot_model` takes two optional arguments:
+`plot_model` takes four optional arguments:
 
 - `show_shapes` (defaults to False) controls whether output shapes are shown in the graph.
 - `show_layer_names` (defaults to True) controls whether layer names are shown in the graph.
+- `expand_nested` (defaults to False) conctrols whether to expand wrapped models into clusters in the graph.
```
Comment (fchollet) [inline docs/templates/visualization.md]: "to expand nested models." seems clear enough

File: keras/utils/vis_utils.py [hunk 2]
```diff
@@ -107,16 +132,21 @@ def model_to_dot(model,
             node_key = layer.name + '_ib-' + str(i)
             if node_key in model._network_nodes:
                 for inbound_layer in node.inbound_layers:
-                    inbound_layer_id = str(id(inbound_layer))
-                    dot.add_edge(pydot.Edge(inbound_layer_id, layer_id))
+                    if not expand_nested or not \
```
Comment (fchollet) [inline keras/utils/vis_utils.py]: Nit: avoid using backslashes to break lines. You can just introduce line breaks on logical operator (wrap the statement inside parens).

File: keras/utils/vis_utils.py [hunk 3]
```diff
@@ -45,35 +48,57 @@ def model_to_dot(model,
             a string specifying the format of the plot:
             'TB' creates a vertical plot;
             'LR' creates a horizontal plot.
+        expand_nested: whether to expand wrapped models into clusters.
+        dpi: dot DPI.
+        subgraph: is this model are wrapped model,
```
Comment (fchollet) [inline keras/utils/vis_utils.py]: "Whether to return a `pydot.Cluster` instance"

File: keras/utils/vis_utils.py [hunk 4]
```diff
@@ -45,35 +48,57 @@ def model_to_dot(model,
...
+            in this case method will return `pydot.Cluster` instance
 
     # Returns
-        A `pydot.Dot` instance representing the Keras model.
+        A `pydot.Graph` instance representing the Keras model.
```
Comment (fchollet) [inline keras/utils/vis_utils.py]: Edit this to cover `subgraph=True` case

**Comment by yoks** [comment]
Added cluster label:
model3

**Comment by gabrieldemarmiesse** [comment]
There seem to be a problem with travis being stuck. Restarting the build by closing and opening again.

**Comment by gabrieldemarmiesse** [comment]
@yoks There is an issue with pep8. A line too long apparently. Could you fix it? Thanks

**Comment by yoks** [comment]
Done, forgot to run check after merging from upstream.
